consoles/sshVirtsh.pm

```diff
@@ -399,8 +399,9 @@ sub _copy_image_to_vm_host ($self, $args, $vmware_openqa_datastore, $file, $name
     }
     else {
         $file = $basedir . $file;
-        # requested expected value in GBs, if there is any passed as an argument
-        my $size = $args->{size} // 0;
+        # requested expected value in GiBs, if there is any passed as an argument
+        my $size = $args->{size} // '0G';
+        $size =~ tr/.*G$/.*/d;
```

Comment:
so you add a G as default just to remove it again in the second line and then in three lines below the G is added again? How about

```suggestion
        $args->{size} =~ tr/.*G$/.*/d;
        my $size = $args->{size} // 0;
```

or

```suggestion
        my $size = $args->{size} =~ tr/.*G$/.*/dr // 0;
```

Reply:
I like suggestion two and amended it accordingly.

Reply:
On a second though, I do think that the original approach was better because it shows the expected format of the `$args->{size}` argument:

```
        my $size = $args->{size} // '0G';
        $size =~ tr/.*G$/.*/d;
```
From the first line one can immediately deduce that the expected format of `$args->{size}` is something like `20G`.

The new solution is more compact, but this information is now also kinda difficult to read out of the context. It's encoded in the regex, but hidden in logic, which makes the code less readable.

Reply:
well, why would the code later add a 'G' as suffix then?

Reply:
This discussion suggests actually to add an explanatory comment, that might resolve this issue. Give me a moment ....

Reply:
@okurz please check again, this is a solution I would be also happy with.